src/extension.ts

```diff
@@ -508,6 +517,34 @@ class Device extends vscode.TreeItem {
             [this.shellPort.toString()]);
         term.show();
     }
+    
+    async captureScreenshot() {
+        const error = (message) => {
+            vscode.window.showErrorMessage("Error capturing screenshot: " + message);
+        }
+        
+        try {
+            const screenshotDirectory = await getSharedTempDir('ev3dev-screenshots');
+            const screenshotFile = `${screenshotDirectory}/ev3dev-${sanitizedDateString()}.png`;
+
+            const conn = await this.exec('fbgrab -');
+            const writeStream = fs.createWriteStream(screenshotFile);
+            writeStream.on('open', () => {
+                conn.stdout.pipe(writeStream);
+            });
+
+            writeStream.on('close', () => {
+                vscode.window.setStatusBarMessage(`Screenshot successfully captured`);
+                vscode.commands.executeCommand('vscode.open', vscode.Uri.file(screenshotFile));
+
+                // Delay to allow viewer to open
+                setTimeout(() => vscode.commands.executeCommand('workbench.action.files.saveAs'), 100);

```

Comment:
> I'm thinking maybe we just display the file and users can click File > Save As... if they want to save it somewhere else.

Now that we know the "save as" option works, I think this would be pretty nice.

> Also, what happens to the temporary file? It kind of looks like we might be tracking them, but it isn't quite obvious. Are the files being deleted when the extension unloads?

Yeah, the theory behind `node-temp` is that once you call `track` any files that were created will be deleted once the process exits (IDE reloads or closes). I haven't validated that this happens properly in the VSCode environment yet.